feat(db): Add database schema, entities, and seeder

3EZYY opened the pull request:
This PR completes Issue #2.

- Adds all 7 database migration files (`up` and `down`).
- Defines all 7 domain entities (structs) in `internal/domain/entity/`.
- Includes the `dbdiagram.io` schema image.
- Adds a seeder script (`cmd/seed/main.go`) to populate dummy data for `users` and `programs`.

Closes #2

3EZYY committed:
feat(db): add all migrations, entities, diagram, and seeder script

3EZYY committed:
feat(db): add all migrations, entities, diagram, and seeder script

ahmdyaasiin commented on cmd/api/seed/main.go:
```diff
@@ -0,0 +1,110 @@
+// seedAdminUser creates a dummy admin user
+func seedAdminUser(authRepo repository.AuthRepository) error {
+	// Hash password
+	hashedPassword, err := hash.HashPassword("password123")
```
you can get the password from the .env file instead of hardcoding it

ahmdyaasiin commented on internal/domain/entity/article.go:
```diff
@@ -0,0 +1,16 @@
+import "time"
+
+// Article represents an article entity in the system
+type Article struct {
+	ID          string     `db:"id" json:"id"`
+	Title       string     `db:"title" json:"title"`
+	Slug        string     `db:"slug" json:"slug"`
+	Content     string     `db:"content" json:"content"`
+	ImageURL    *string    `db:"image_url" json:"image_url,omitempty"`
+	AuthorID    *string    `db:"author_id" json:"author_id,omitempty"`
+	PublishedAt *time.Time `db:"published_at" json:"published_at,omitempty"`
+	CreatedAt   time.Time  `db:"created_at" json:"created_at"`
+	UpdatedAt   time.Time  `db:"updated_at" json:"updated_at"`
+}
```
you can split this struct into an entity and a DTO so the entity struct will have db tags and the DTO struct will have the json tags

ahmdyaasiin replied:
do this for all entity structs

3EZYY committed:
chore: update go.mod and go.sum

3EZYY committed:
fix(review): apply feedback (separate DTO/Entity and update seeder)

3EZYY commented:
Refactor: Split Entity/DTO (User, Article) and move Seeder password to Config

This PR refactors the config and domain layers to improve security and separation of concerns, fulfilling the feedback from the review.

- Main Changes
* Config & Seeder:
Moved the hardcoded admin password from the seeder (cmd/seed/main.go) to the .env file as SEEDER_ADMIN_PASS.
Updated internal/config/config.go and cmd/seed/main.go to load and use this new variable.

* Domain (Entity/DTO Split):
Split Article and User structs as requested.
Entity (internal/domain/entity/): Structs for Article and User now only contain db:"..." tags.
DTO (internal/domain/dto/): Created new ArticleResponse and UserResponse structs that only contain json:"..." tags.
Added converter.go with helper functions (e.g., ToUserResponse, ToArticleResponse) to map Entities to DTOs.

* ⚠️ Breaking Change / Next Steps
This PR only creates the DTOs. The handlers (e.g., in internal/auth/handler/) have not been updated yet. They still return the Entity structs.
A follow-up PR is required to update all handlers to use the converters and return the new DTOs (e.g., return dto.ToUserResponse(user)), otherwise all JSON API responses will be broken.

3EZYY committed:
fix(review): apply feedback (separate DTO/Entity and update seeder)

3EZYY commented:
This PR completes **Issue #2** and applies all feedback from the latest review.

All changes are focused *only* on the database, entity, and seeder layers. All `Issue #3` files (handlers, services, api `main.go`) are excluded from this PR.

### 1. Seeder Refactor (Feedback)
* Moved the hardcoded admin password from `cmd/seed/main.go`.
* Added `SEEDER_ADMIN_PASS` to `internal/config/config.go` and updated the seeder to use this config variable.

### 2. Entity/DTO Separation (Feedback)
* **Entities** (`internal/domain/entity/`):
    * Removed all `json:"..."` tags from **all 7 entity structs** (`User`, `Article`, `Program`, `Event`, `Donation`, `Volunteer`, `RefreshToken`).
    * These structs now **only** contain `db:"..."` tags, as requested.
* **DTOs** (`internal/domain/dto/`):
    * Created 5 new DTO files (`program_dto.go`, `event_dto.go`, etc.) to complete the set.
    * All DTO `...Response` structs now **only** contain `json:"..."` tags.
* **Converters** (`internal/domain/dto/converter.go`):
    * Added converter functions (e.g., `ToProgramResponse`, `ToEventResponse`, `ToProgramResponseList`, etc.) for all the new DTOs.

This PR is now ready for re-review.

Closes #2
